spec/habitat_spec.cr

```diff
@@ -37,6 +37,21 @@ class Child < Parent
   end
 end
 
+class WithSpecialFormat
+  Habitat.create do
+    setting pin : String, validation: :pin_format
+    setting code : Int32, validation: :code_format
+  end
+
+  def self.pin_format(value : String)
+    value.match(/^\d{4}$/) || settings.invalid("Number must be exactly 4 digits")
+  end
+
+  def self.code_format(value : Int32)
+    (value > 99 && value < 199) || settings.invalid("Number must between 99 and 199")
```

Comment:
I think it would be nice to clarify that this will raise an error. When I was first reading this I thought it was maybe adding an error to an errors object. 

What if instead the API was `Habitat.raise_validation_error("My error message")`

A tad bit longer, but I think it is more clear how to look up documentation and its clear it stops executation and raises. What do you think?

Reply:
I like that idea.